scripts/lib/install-node

```diff
@@ -21,9 +21,14 @@ if [ "$($YARN_BIN --version 2>/dev/null)" = "$yarn_version" ] && [ "$current_nod
 fi
 
 if [ "$current_node_version" != "v$node_version" ]; then
-    export NVM_DIR=/usr/local/nvm
+    if [ CIRCLECI ]; then
+        export NVM_DIR="$HOME/nvm"
+    else
+        export NVM_DIR=/usr/local/nvm
+    fi
```

Comment:
Can you explain why this bit should be different in CircleCI from elsewhere? Is this to match some other bit of our code doing it differently, or is there something else going on?

I know earlier you were getting a permissions error at some point, but that was for a path that doesn't seem obviously related to this one (and also I'd want to make sure we were fixing the issue in the right direction -- e.g. making the one bit of code that's behaving differently from everything else fall into line, rather than make a second piece of code follow its example.)

If the root issue is that the `circleci` user doesn't have certain permissions in a way that differs from our dev environment, or something like that, then that's easy to fix directly -- unlike on Travis, we control the base image completely. Take a look at `tools/circleci/Dockerfile.template`.

Reply:
For some reason `/usr/local/nvm` is not accessible but current circleci user.

Reply:
And one more thing i forgot to mention i figured out that installing in `/usr/local/nvm` was a problem when i tried in into a ssh build it failes which gave me some clue that we need to change this.

Reply:
> For some reason /usr/local/nvm is not accessible but current circleci user.
 
It's not writable by unprivileged users in the dev environment either, at least not in my dev VM:

```
$ ls -ld /usr/local/nvm/
drwxr-xr-x 7 root root 4096 Jan 20 01:31 /usr/local/nvm/
```

How does this work in the dev environment -- presumably by using `sudo` somewhere? Whatever we do there, we should do the same thing in CI. If we let things differ unnecessarily between dev and CI, it just becomes an endless source of hard-to-diagnose bugs like this one -- so we should squash the differences instead.